apps/fz_http/lib/fz_http_web/auth/json/authentication.ex

```diff
@@ -0,0 +1,28 @@
+defmodule FzHttpWeb.Auth.JSON.Authentication do
+  @moduledoc """
+  API Authentication implementation module for Guardian.
+  """
+  use Guardian, otp_app: :fz_http
+
+  alias FzHttp.{
+    ApiTokens,
+    Users
+  }
+
+  @impl Guardian
+  def subject_for_token(user, _claims) do
+    {:ok, user.id}
+  end
+
+  @impl Guardian
+  def resource_from_claims(%{"jti" => api_token_id}) do
+    with {:ok, api_token} <- ApiTokens.get_api_token(api_token_id),
```

Comment:
By Elixir conventions functions starting with `get_` return `term()` or `nil`, while `fetch_` return `{:ok, term()}` or `:error` tuple (can be `{:error, reason}`). Should we adopt it?

Reply:
I'm open to what you think is best. If we're deviating from the Phoenix defaults I think another strong pattern would be a viable candidate.

Reply:
I will admit the `{:ok, term()}` `{:error, reason}` pattern fits nicely with the `with/1` expression.

Reply:
EDIT: This was simply a bug I introduced, I'll update the `with/1` to match this properly.